src/Shell/Task/ModelTask.php

```diff
@@ -273,10 +273,17 @@ public function findBelongsTo($model, array $associations)
                         $tmpModelName = Inflector::camelize($found);
                     }
                 }
+                $foreignKeys = (array) TableRegistry::get($tmpModelName)->primaryKey();
+                foreach ($foreignKeys as $k => $v) {
+                    if ($v === 'id') {
```

Comment:
I ll try to get this PR through. If I do I'll take a hard look at teaching bake to (optionally) respect real foreign keys, so that you can opt-out of dilemmas involving non-conventional primary/foreign (composite) keys.